mkosi/installer/apk.py

```diff
@@ -0,0 +1,121 @@
+            "mkosi-upgrade":   ["apk", "--update-cache", "upgrade"],
+            "mkosi-remove":    ["apk", "--remove", "del"],
+            "mkosi-reinstall": ["apk", "--update-cache", "fix"],
+        }  # fmt: skip
+
+    @classmethod
+    def mounts(cls, context: Context) -> list[PathString]:
+        mounts: list[PathString] = [
+            *super().mounts(context),
+        ]
+
+        mounts += ["--ro-bind", "/etc/apk/keys", "/etc/apk/keys"]
+
+        cache_dir = context.config.package_cache_dir_or_default() / "apk"
+        cache_dir.mkdir(parents=True, exist_ok=True)
+        mounts += ["--bind", cache_dir, "/var/cache/apk"]
+
+        return mounts
+
+    @classmethod
+    def setup(cls, context: Context, repositories: Iterable[str]) -> None:
+        # Create cache symlink to enable automatic caching
+        cache_link = context.root / "etc/apk/cache"
+        if not cache_link.exists():
+            cache_link.parent.mkdir(exist_ok=True, parents=True)
+            cache_link.symlink_to("/var/cache/apk")
+
+        config = context.root / "etc/apk/repositories"
+        if config.exists():
+            print("apk setup() repo file exists!")
+            return
+
+        config.parent.mkdir(exist_ok=True, parents=True)
+
+        with config.open("w") as f:
+            for repo in repositories:
+                f.write(f"{repo}\n")
+
+    @classmethod
+    def cmd(cls, context: Context) -> list[PathString]:
+        return [
+            "apk",
+            "--keys-dir", "/etc/apk/keys",
+            "--root", "/buildroot",
+            "--cache-dir", "/var/cache/apk",
+            "--arch", context.config.distribution.architecture(context.config.architecture),
+            "--no-interactive",
+            "--update-cache",
+            *(["--allow-untrusted"] if not context.config.repository_key_check else []),
+        ]  # fmt: skip
+
+    @classmethod
+    def invoke(
+        cls,
+        context: Context,
+        operation: str,
+        arguments: Sequence[str] = (),
+        *,
+        apivfs: bool = False,
+        stdout: _FILE = None,
+    ) -> CompletedProcess:
+        return run(
+            cls.cmd(context) + [operation, *arguments],
+            sandbox=cls.sandbox(context, apivfs=apivfs),
+            env=cls.finalize_environment(context),
+            stdout=stdout,
+        )
+
+    @classmethod
+    def install(
+        cls,
+        context: Context,
+        packages: Sequence[str],
+        *,
+        apivfs: bool = True,
+    ) -> None:
+        cls.invoke(context, "add", packages, apivfs=apivfs)
+
+    @classmethod
+    def remove(cls, context: Context, packages: Sequence[str]) -> None:
+        cls.invoke(context, "del", packages, apivfs=True)
+
+    @classmethod
+    def sync(cls, context: Context, force: bool) -> None:
+        # TODO implement force
+        if (context.root / "etc/apk/world").exists():
+            cls.invoke(context, "update", [])
```

Comment:
This needs some comments, I have no clue why the file would need to exist in the rootfs to be able to sync repository metadata.